### PR TITLE
[FSDP][Easy] Fix context manager syntax

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#91410 [FSDP][Easy] Fix context manager syntax**
* #91044 [FSDP][7/N] Support `replicate` in `fully_shard`
* #90959 [FSDP][6/N] Add note explaining idioms for `_FSDPState` traversal
* #90874 [FSDP][5/N] Add manual "wrapping" support for `fully_shard`
* #90871 [FSDP][4/N] Refactor func to share state/init handle attrs
* #90945 fully_shard load state_dict

### CODE REVIEW

**Comment by kit1980** [comment]
Let's wait with merging this until we understand what's going on with https://github.com/pytorch/pytorch/pull/91371 so we don't mask the symptoms.

**Review by kit1980 — APPROVED** [review]
